src/plugins/payments-stripe/util/getStripeInstanceForShop.js

```diff
@@ -13,8 +14,8 @@ export default async function getStripeInstanceForShop(context, shopId) {
   const stripePkg = await getStripePackageForShop(context, shopId);
   const stripePkgSettings = (stripePkg || {}).settings || {};
 
-  const stripeApiKey = stripePkgSettings.api_key;
-  if (!stripeApiKey) {
+  const { STRIPE_API_KEY } = config;
+  if (!STRIPE_API_KEY) {
     const stripeAccessToken = (stripePkgSettings.connectAuth || {}).access_token;
```

Comment:
This part looking for `connectAuth` and the `if (stripeAccessToken)` block that follows should be removed, too. You could really remove this whole `if (!STRIPE_API_KEY)` block now because envalid is checking the ENV for you.

Reply:
That will also mean that you can avoid calling `getStripePackageForShop` above.